### PR TITLE
[AIRFLOW-2389] Create a pinot db api hook

### PR DESCRIPTION
**Pull request by feng-tao**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2389
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
  create a dbapi hook to query pinot(https://github.com/linkedin/pinot)

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
  yes

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/hooks/pinot_hook.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    def get_records(self, sql):
+        """
+        Executes the sql and returns a set of records.
+
+        :param sql: the sql statement to be executed (str) or a list of
+            sql statements to execute
+        :type sql: str
+        """
+        if sys.version_info[0] < 3:
```
Comment (Fokko) [inline airflow/contrib/hooks/pinot_hook.py]: Could you use six here? https://pythonhosted.org/six/

**Review by Fokko — CHANGES_REQUESTED** [review]
Nice addition, one minor remark. Also the tests are really minimal, would be nice to actually build a connection string.

File: airflow/contrib/hooks/pinot_hook.py (hunk 1 above)
Comment (feng-tao) [reply airflow/contrib/hooks/pinot_hook.py]: Thanks for the suggestion @Fokko , just add another test. PR is updated.

**Review by Fokko — APPROVED** [review]
Thanks @feng-tao looks good now. Cheers

**Pull request closed without merging**